src/librustc_trans/debuginfo/namespace.rs

```diff
@@ -63,13 +63,21 @@ pub fn item_namespace(ccx: &CrateContext, def_id: DefId) -> DIScope {
         })
     });
 
-    let namespace_name = match def_key.disambiguated_data.data {
-        DefPathData::CrateRoot => ccx.tcx().crate_name(def_id.krate).as_str(),
-        data => data.as_interned_str()
+    let (namespace_name, span) = match def_key.disambiguated_data.data {
+        DefPathData::CrateRoot => {
+            // We don't provide the span for the crate root. Because of
+            // implementation reasons it is not possible to access this
+            // information without introducing a dep-tracking edge to
+            // DepNode::Krate, which essentially breaks incr. comp.
+            (ccx.tcx().crate_name(def_id.krate).as_str(), DUMMY_SP)
```

Comment:
Well I would point it at `DepNode::Hir(CRATE_DEF_ID)`, I think.

Note that DepNode::Krate and the Hir node for the crate def-id are distinct nodes.